plugins/flytekit-pydantic/flytekitplugins/pydantic/serialization.py

```diff
@@ -0,0 +1,119 @@
+"""
+Logic for serializing a basemodel to a literalmap that can be passed between flyte tasks.
+
+The serialization process is as follows:
+
+1. Serialize the basemodel to json, replacing all flyte types with unique placeholder strings
+2. Serialize the flyte types to separate literals and store them in the flyte object store (a singleton object)
+3. Return a literal map with the json and the flyte object store represented as a literalmap {placeholder: flyte type}
+
+"""
+import uuid
+from typing import Any, Dict, Union, cast
+
+import pydantic
+from google.protobuf import struct_pb2
+from typing_extensions import Annotated
+
+from flytekit.core import context_manager, type_engine
+from flytekit.models import literals
+
+from . import commons
+
+BASEMODEL_JSON_KEY = "BaseModel JSON"
+FLYTETYPE_OBJSTORE_KEY = "Flyte Object Store"
+
+
+SerializedBaseModel = Annotated[str, "A pydantic BaseModel that has been serialized with placeholders for Flyte types."]
+
+ObjectStoreID = Annotated[str, "Key for unique literalmap of a serialized basemodel."]
+LiteralObjID = Annotated[str, "Key for unique object in literal map."]
+LiteralStore = Annotated[Dict[LiteralObjID, literals.Literal], "uid to literals for a serialized BaseModel"]
+
+
+class BaseModelFlyteObjectStore:
+    """
+    This class is an intermediate store for python objects that are being serialized/deserialized.
+
+    On serialization of a basemodel, flyte objects are serialized and stored in this object store.
+    """
+
+    def __init__(self) -> None:
+        self.literal_store: LiteralStore = dict()
+
+    def register_python_object(self, python_object: object) -> LiteralObjID:
+        """Serialize to literal and return a unique identifier."""
+        serialized_item = serialize_to_flyte_literal(python_object)
+        identifier = make_identifier_for_serializeable(python_object)
+        self.literal_store[identifier] = serialized_item
```

Comment:
Could we please check here that `identifier` isn't already contained in `literal_store`.